x/cronos/keeper/keeper.go

```diff
@@ -263,7 +256,7 @@ func (k Keeper) RegisterOrUpdateTokenMapping(ctx sdk.Context, msg *types.MsgUpda
 		k.bankKeeper.SetDenomMetaData(ctx, metadata)
 
 		// update the mapping
-		if err := k.SetExternalContractForDenom(ctx, msg.Denom, common.HexToAddress(contract)); err != nil {
+		if err := k.SetExternalContractForDenom(ctx, msg.Denom, common.HexToAddress(msg.Contract)); err != nil {
```

Comment:
Is this change to allow map cronos originated token to arbitrary denom?

Reply:
yes and add some integration tests.

to be able to set a proxy contract for old source crc20 tokens, we need to be able to allow any contract address for a specific denom